### PR TITLE
removed vep_filters parameter will now be generated from clinical panel

### PR DESCRIPTION
**Pull request by Lucpen**
## PR checklist

- [x] removed vep_filters parameter will now be generated from clinical panel
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/genomic-medicine-sweden/tomte/tree/master/.github/CONTRIBUTING.md)
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: subworkflows/local/annotate_snv.nf [hunk 1]
```diff
@@ -28,17 +35,52 @@ workflow ANNOTATE_SNV {
...
+        ch_versions = ch_versions.mix( GAWK.out.versions )
+        ch_versions = ch_versions.mix( ENSEMBLVEP_FILTERVEP.out.versions )
+        ch_versions = ch_versions.mix( TABIX_BGZIPTABIX.out.versions )
+        ch_versions = ch_versions.mix( TABIX_TABIX.out.versions )
 
     emit:
-        vcf_gz   = ENSEMBLVEP_VEP.out.vcf    // channel: [ val(meta), path(vcf.gz) ]
-        tbi_gz   = TABIX_VEP.out.tbi         // channel: [ val(meta), path(tbi) ]
+        //vcf_gz   = ENSEMBLVEP_VEP.out.vcf    // channel: [ val(meta), path(vcf.gz) ]
+        //tbi_gz   = TABIX_VEP.out.tbi         // channel: [ val(meta), path(tbi) ]
```
Comment (Lucpen) [inline subworkflows/local/annotate_snv.nf:78]:
```suggestion
```

File: subworkflows/local/annotate_snv.nf [hunk 2]
```diff
@@ -28,17 +35,52 @@ workflow ANNOTATE_SNV {
...
         tab_gz   = ENSEMBLVEP_VEP.out.tab    // channel: [ val(meta), path(tab.gz) ]
         json_gz  = ENSEMBLVEP_VEP.out.json   // channel: [ val(meta), path(json.gz) ]
         report   = ENSEMBLVEP_VEP.out.report // channel: [ path(html) ]
+        ch_vcf_clin = ch_vcf_clin // channel: [ val(meta), path(vcf.gz) path(tbi)]
+        ch_vcf_research = ch_vcf_research // channel: [ val(meta), path(vcf.gz) path(tbi)]
```
Comment (rannick) [inline subworkflows/local/annotate_snv.nf:83]: Indentation

**Review by rannick — APPROVED** [review]
LGTM!

File: subworkflows/local/annotate_snv.nf [hunk 3]
```diff
@@ -28,17 +35,52 @@ workflow ANNOTATE_SNV {
...
         versions = ch_versions               // channel: [ path(versions.yml) ]
 }
```
Comment (Lucpen) [inline subworkflows/local/annotate_snv.nf:86]:
```suggestion
        tab_gz                  = ENSEMBLVEP_VEP.out.tab       // channel: [ val(meta), path(tab.gz) ]
        json_gz                = ENSEMBLVEP_VEP.out.json     // channel: [ val(meta), path(json.gz) ]
        report                   = ENSEMBLVEP_VEP.out.report  // channel: [ path(html) ]
        ch_vcf_clin          = ch_vcf_clin                                 // channel: [ val(meta), path(vcf.gz) path(tbi)]
        ch_vcf_research = ch_vcf_research                       // channel: [ val(meta), path(vcf.gz) path(tbi)]
        versions                = ch_versions                               // channel: [ path(versions.yml) ]
}

```

File: subworkflows/local/annotate_snv.nf [hunk 4]
```diff
@@ -28,17 +35,52 @@ workflow ANNOTATE_SNV {
...
```
Comment (Lucpen) [inline subworkflows/local/annotate_snv.nf:85]:
```suggestion
        tab_gz          = ENSEMBLVEP_VEP.out.tab    // channel: [ val(meta), path(tab.gz) ]
        json_gz         = ENSEMBLVEP_VEP.out.json   // channel: [ val(meta), path(json.gz) ]
        report          = ENSEMBLVEP_VEP.out.report // channel: [ path(html) ]
        ch_vcf_clin     = ch_vcf_clin               // channel: [ val(meta), path(vcf.gz) path(tbi)]
        ch_vcf_research = ch_vcf_research           // channel: [ val(meta), path(vcf.gz) path(tbi)]
        versions        = ch_versions               // channel: [ path(versions.yml) ]
}
```

**Review by jemten — COMMENTED** [review]
I might be missing something but it looks like the gawk process and the CREATE_HGNCIDS_FILE processes are trying to do the same thing.

Could you also update the ensembl/vep version as well so that we are on the same version as filter_vep?

File: subworkflows/local/annotate_snv.nf [hunk 5]
```diff
@@ -28,17 +35,50 @@ workflow ANNOTATE_SNV {
...
+        TABIX_TABIX( RENAME_FILES.out.output )
+        ch_vcf_research = RENAME_FILES.out.output.join(TABIX_TABIX.out.tbi)
+
+        GAWK( ch_gene_panel_clinical_filter.map{it -> [[id:'hgnc'], it]}.collect(),
+            [] )
+
+        // Generate Clinical filter
+        CREATE_HGNCIDS_FILE( GAWK.out.output )
+            .txt
+            .set {ch_hgnc_ids}
```
[the remark this answers is not in the document]
Comment (Lucpen) [reply subworkflows/local/annotate_snv.nf:57]: You are right, it was a misunderstanding of `CREATE_HGNCIDS_FILE` XD

**Review by jemten — APPROVED** [review]
👍

**Pull request merged**